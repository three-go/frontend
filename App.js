import React, { useEffect } from "react";

import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import SplashScreen from "react-native-splash-screen";

import { GameContextProvider } from "./src/context/GameContext";
import FaceGoContainer from "./src/features/faceGo/container/FaceGoContainer";
import MainContainer from "./src/features/main/container/MainContainer";
import ShoutGoContainer from "./src/features/shoutGo/container/ShoutGoContainer";
import TimmerContainer from "./src/features/shoutGo/container/TimerContainer";

const Stack = createNativeStackNavigator();

const App = () => {
  useEffect(() => {
    if (SplashScreen.hide) {
      SplashScreen.hide();
    }
  }, []);

  return (
    <NavigationContainer>
      <GameContextProvider>
        <Stack.Navigator initialRouteName="Main">
          <Stack.Screen
            name="Main"
            component={MainContainer}
            options={{ headerShown: false }}
          />

          <Stack.Screen
            name="FaceGo"
            component={FaceGoContainer}
            options={{ headerShown: false }}
          />

          <Stack.Screen
            name="ShoutGo"
            component={ShoutGoContainer}
            options={{ headerShown: false }}
          />

          <Stack.Screen
            name="Timer"
            component={TimmerContainer}
            options={{ headerShown: false }}
          />
        </Stack.Navigator>
      </GameContextProvider>
    </NavigationContainer>
  );
};

export default App;
