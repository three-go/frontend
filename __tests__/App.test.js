import React from "react";

import { render } from "@testing-library/react-native";
import "react-native-gesture-handler/jestSetup";
import "@testing-library/jest-native";

import App from "../App";

const mockNavigate = jest.fn();
const mockDisPatch = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockNavigate,
      dispatch: mockDisPatch,
    }),
  };
});

jest.mock("react-native-reanimated", () => {
  const Reanimated = require("react-native-reanimated/mock");
  Reanimated.default.call = () => {};
  return Reanimated;
});

jest.mock("react-native/Libraries/EventEmitter/NativeEventEmitter.js", () =>
  require("react-native/Libraries/EventEmitter/__mocks__/NativeEventEmitter.js")
);

jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");

jest.mock("react-native-sound", () => {
  const Sound = () => {
    const obj = {};
    obj.sound = "music";
    obj.setNumberOfLoops = jest.fn();
    obj.play = jest.fn();
    return obj;
  };

  return Sound;
});

jest.mock("@react-native-community/async-storage", (key, value) => {
  const storage = {};
  storage[key] = value;
  return storage;
});

jest.mock("react-native-permissions", () =>
  require("react-native-permissions/mock")
);

jest.mock("react-native-splash-screen", () => {
  const Module = () => {
    const screen = {
      default: {
        hide: jest.fn(),
      },
    };
    return screen;
  };

  return Module;
});

describe("Component Test", () => {
  it("App Test", () => {
    const app = render(<App />);
    const appJson = app.toJSON();
    const main = app.getByTestId("mainContainer");
    const faceGo = app.getByText("Face Go");
    const ShoutGo = app.getByText("Shout Go");
    const score = app.getByText("점수 보기");
    const exit = app.getByText("게임 종료");

    expect(main).toBeTruthy();
    expect(faceGo).toHaveTextContent("Face Go");
    expect(ShoutGo).toHaveTextContent("Shout Go");
    expect(score).toHaveTextContent("점수 보기");
    expect(exit).toHaveTextContent("게임 종료");
    expect(appJson).toBeTruthy();
    expect(appJson).toMatchSnapshot();
  });
});
