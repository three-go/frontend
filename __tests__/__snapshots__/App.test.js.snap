// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Test App Test 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Main"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RNCSafeAreaProvider
          onInsetsChange={[Function]}
          style={
            Array [
              Object {
                "flex": 1,
              },
              undefined,
            ]
          }
          testID="mainContainer"
        >
          <RNCSafeAreaView
            edges={
              Array [
                "bottom",
                "left",
                "right",
                "top",
              ]
            }
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#212529",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View>
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "testUri": "../../../public/assets/logo.png",
                  }
                }
                style={
                  Object {
                    "height": 160,
                    "width": 240,
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "height": 100,
                  "justifyContent": "space-evenly",
                  "marginBottom": 100,
                  "marginTop": 50,
                  "width": "100%",
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#00BBD1",
                    "borderColor": "#FCF8F6",
                    "borderRadius": 4,
                    "borderWidth": 4,
                    "height": 120,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": 120,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FCF8F6",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "padding": 8,
                    }
                  }
                >
                  Face Go
                </Text>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#00E0EA",
                    "borderColor": "#FCF8F6",
                    "borderRadius": 4,
                    "borderWidth": 4,
                    "height": 120,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": 120,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FCF8F6",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "padding": 8,
                    }
                  }
                >
                  Shout Go
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "marginBottom": 20,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#96A1A8",
                    "borderRadius": 4,
                    "height": 40,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": 150,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FCF8F6",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "padding": 8,
                    }
                  }
                >
                  점수 보기
                </Text>
              </View>
            </View>
            <View
              style={
                Object {
                  "marginBottom": 20,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#f45a5a",
                    "borderRadius": 4,
                    "height": 40,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": 150,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "#FCF8F6",
                      "fontSize": 15,
                      "fontWeight": "bold",
                      "padding": 8,
                    }
                  }
                >
                  게임 종료
                </Text>
              </View>
            </View>
          </RNCSafeAreaView>
        </RNCSafeAreaProvider>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
